source/future.tex

```diff
@@ -98,20 +98,58 @@
 prefix \tcode{++} and \tcode{\dcr} expressions\iref{expr.pre.incr}
 of \tcode{volatile}-qualified arithmetic and pointer types are deprecated.
 
+\begin{example}
+\begin{codeblock}
+volatile int velociraptor;
+++velociraptor;                 // deprecated: is this a single memory operation, two, or a six-foot turkey?
+\end{codeblock}
+\end{example}
+
+
 \pnum
 Certain assignments
 where the left operand is a \tcode{volatile}-qualified non-class type
 are deprecated; see~\ref{expr.ass}.
 
+\begin{example}
+\begin{codeblock}
+int neck, tail;
+volatile int brachiosaur;
+tail = brachiosaur = neck;      // deprecated: does this touch brachiosaur once or twice?
+\end{codeblock}
+\end{example}
+
+
 \pnum
 A function type\iref{dcl.fct}
 with a parameter with \tcode{volatile}-qualified type or
 with a \tcode{volatile}-qualified return type is deprecated.
 
+\begin{example}
+\begin{codeblock}
+// Deprecated: a volatile return type has no meaning to the caller.
+volatile amber jurassic();
+// The volatile left_arm and right_arm parameter are useless to the caller.
+void trex(volatile short left_arm, volatile short right_arm);
+\end{codeblock}
+\end{example}
+
+
 \pnum
 A structured binding\iref{dcl.struct.bind} of a \tcode{volatile}-qualified type
 is deprecated.
 
+\begin{example}
+\begin{codeblock}
+struct linhenykus { volatile short forelimb; };
+void (linhenykus alvarezsauroid) {
+  auto& [what_is_this] = alvarezsauroid; // deprecated: very confusing
```

Comment:
This should maybe have a more specific description than "very confusing". If we're going to start deprecating C++ constructs that are "very confusing", we're going to have a *ton* of work to do.